api/src/paths/dwc/scrape-occurrences.ts

```diff
@@ -0,0 +1,186 @@
+import { getDBConnection } from '../../database/db';
+import { RequestHandler } from 'express';
+import { Operation } from 'express-openapi';
+import { SYSTEM_ROLE } from '../../constants/roles';
+import { getLogger } from '../../utils/logger';
+import { logRequest } from '../../utils/path-utils';
+import { getSubmissionFileFromS3, getSubmissionS3Key, prepDWCArchive } from './validate';
+import { HTTP400 } from '../../errors/CustomError';
+import { postOccurrenceSQL } from '../../queries/occurrence/occurrence-create-queries';
+import { PostOccurrence } from '../../models/occurrence-create';
+import xlsx from 'xlsx';
+import moment from 'moment';
+
+const defaultLog = getLogger('paths/dwc/scrape-occurrences');
+
+export const POST: Operation = [
+  logRequest('paths/dwc/scrape-occurrences', 'POST'),
+  getSubmissionS3Key(),
+  getSubmissionFileFromS3(),
+  prepDWCArchive(),
+  scrapeAndUploadOccurrences()
+];
+
+POST.apiDoc = {
+  description: 'Scrape information from file into occurrence table.',
+  tags: ['scrape', 'occurrence'],
+  security: [
+    {
+      Bearer: [SYSTEM_ROLE.SYSTEM_ADMIN, SYSTEM_ROLE.PROJECT_ADMIN]
+    }
+  ],
+  requestBody: {
+    description: 'Request body',
+    content: {
+      'application/json': {
+        schema: {
+          type: 'object',
+          required: ['occurrence_submission_id'],
+          properties: {
+            occurrence_submission_id: {
+              description: 'A survey occurrence submission ID',
+              type: 'number',
+              example: 1
+            }
+          }
+        }
+      }
+    }
+  },
+  responses: {
+    200: {
+      description: 'Successfully scraped and uploaded occurrence information.'
+    },
+    400: {
+      $ref: '#/components/responses/400'
+    },
+    401: {
+      $ref: '#/components/responses/401'
+    },
+    403: {
+      $ref: '#/components/responses/401'
+    },
+    500: {
+      $ref: '#/components/responses/500'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+export function scrapeAndUploadOccurrences(): RequestHandler {
+  return async (req, res, next) => {
+    defaultLog.debug({ label: 'scrapeAndUploadOccurrences', message: 'params', files: req.body });
+
+    const occurrenceSubmissionId = req.body.occurrence_submission_id;
+    const file = req['dwcArchive'];
+
+    const connection = getDBConnection(req['keycloak_token']);
+
+    try {
+      await connection.open();
+
+      const eventHeaders = file.worksheets.event?.getHeaders();
```

Comment:
I thought we had common code for the validate endpoints that get the sheets, headers and rows.  Can we leverage that?

Reply:
which one?

Reply:
one sec ... lets go to discord